change sysadmin dashboard navigation w.r.t edx-sysadmin plugin and remove old sysadmin code

HamzaIbnFarooq opened the pull request:
### Related tickets
https://github.com/mitodl/edx-sysadmin/issues/1
https://github.com/mitodl/edx-platform/issues/250

### What this PR does
* replace sysadmin navigation w.r.t edx-sysadmin plugin
* removes sysadmin related stuff from edx-platform

### NOTE
This PR is just for internal reviews and testing, once all the requirements are fulfilled, we need to create an upstream PR from this branch.
It's a follow-up of https://github.com/mitodl/edx-platform/pull/251, but now rebased on latest edx-platform master branch.

ziafazal reviewed (approved):
👍

pdpinch commented:
I don't understand what this PR is doing? It seems to be removing the sysadmin dashboard (good!) but then it adds a sysadmin utils file and a sprinkling of conditionals? I don't think edx will accept this in the master branch. Is this for our fork? Is there no way to remove the need for forking?

pdpinch commented on lms/djangoapps/instructor/views/instructor_dashboard.py:
```diff
@@ -429,7 +430,7 @@ def _section_course_info(course, access):
         section_data['enrollment_message'] = message
 
-    if settings.FEATURES.get('ENABLE_SYSADMIN_DASHBOARD'):
+    if is_edx_sysadmin_installed():
```
What's this for? Can it be removed?

pdpinch commented on lms/templates/header/navbar-authenticated.html:
```diff
@@ -53,10 +54,10 @@
       </div>
     % endif
-    % if show_sysadmin_dashboard:
+    % if show_sysadmin:
```
Can we move this to the mitx theme instead of changing edx-platform?

pdpinch commented on lms/templates/instructor/instructor_dashboard_2/course_info.html:
```diff
@@ -114,7 +115,7 @@ <h4 class="hd hd-4">${_("Basic Course Information")}</h4>
   </ul>
 
-  %if settings.FEATURES.get('ENABLE_SYSADMIN_DASHBOARD', '') and user.is_staff:
+  %if show_sysadmin_dashboard(user):
```
Can this be moved to the MITx theme instead?

pdpinch commented on lms/templates/navigation/bootstrap/navbar-authenticated.html:
```diff
@@ -48,10 +49,10 @@
       <li class="nav-item mt-2 nav-item-open-collapsed">
         ## Translators: This is short for "System administration".
-        <a class="nav-link" href="${reverse('sysadmin')}">${_("Sysadmin")}</a>
+        <a class="nav-link" href="${reverse('sysadmin:sysadmin')}">${_("Sysadmin")}</a>
```
Can this be moved to the MITx theme instead?

pdpinch commented on lms/templates/navigation/navbar-authenticated.html:
```diff
@@ -30,10 +31,10 @@
       <li class="item">
         ## Translators: This is short for "System administration".
-        <a class="btn" href="${reverse('sysadmin')}">${_("Sysadmin")}</a>
+        <a class="btn" href="${reverse('sysadmin:sysadmin')}">${_("Sysadmin")}</a>
```
Can this be moved to the MITx theme?

HamzaIbnFarooq commented:
> I don't understand what this PR is doing? It seems to be removing the sysadmin dashboard (good!) but then it adds a sysadmin utils file and a sprinkling of conditionals? I don't think edx will accept this in the master branch. Is this for our fork? Is there no way to remove the need for forking?

@pdpinch, it is not related to our fork or theme, it depends on how we want the navigation to sysadmin to be done.

**Currently**, sysadmin is navigated through a tab in edx-platform's navigation bar:

<img  src="https://user-images.githubusercontent.com/42243411/112997283-6c16a280-9186-11eb-9d1a-b2f6cdfff885.png"  width="400px"/>

If we want the tab to be shown when edx-sysadmin plugin is installed, then we have to put some code (sysadmin_utils) in the edx-platform `to check the availability of edx-sysadmin plugin` and `check if the user is eligible to view that tab`.

I believe, `we can't put a Navigation Tab into edx-platform dynamically through a separate plugin without editing the templates in edx-platform`. So if we want to remove that code from edx-platform then the user has to put link directly into the  browser to access sysadmin. And we won't be able to put a link under instructor tab [regarding git log](https://github.com/mitodl/edx-sysadmin/issues/1#issuecomment-810204618) as well.

pdpinch commented:
Can common/djangoapps/util/sysadmin_utils.py be part of the [syadmin plugin](https://github.com/mitodl/edx-sysadmin/) instead of adding it to edx-platform? They won't accept it in master. 

We already override many of the templates in [mitx-theme ](https://github.com/mitodl/mitx-theme/). I think it will be easier to maintain those changes in that code base, instead of maintaining them in our fork of edx-platform. 

Every time we make a commit to our fork, it becomes a maintenance burden for us, and needs to be ported withe every upgrade. The whole purpose of this sysadmin exercise is to remove this code from our fork.

ziafazal commented:
@pdpinch @HamzaIbnFarooq 
Yes we can move `common/djangoapps/util/sysadmin_utils.py` functions to plugin's view and populate the context of dashboard view in LMS with data we need in the templates. Plugins provide [get_plugins_view_context](https://github.com/edx/edx-django-utils/blob/master/edx_django_utils/plugins/plugin_contexts.py#L15) method to achieve this. However, we'd still have to modify templates in open edX to add sys admin menu on LMS dashboard. 
If we want to avoid changes in templates of open edX we have to enhance current plugins support of open edX to allow defining of dashboard menus in plugins which can be shown on dashboard. If you think we should enhance current plugins support to add ability to define menus in LMS/Studio dashboard lets start a discussion on [community board](https://discuss.openedx.org/).

HamzaIbnFarooq commented:
Thanks @ziafazal,
I agree with your `enhancing plugin support` suggestion, we can opt that solution in the long run. Right now, as @pdpinch  asked, as a first step I have removed everything related to sysadmin from edx-platform in this PR.

Now for the time being, `sysadmin` tab will not be shown to users, but they can directly hit `/sysadmin/` URL to access sysadmin.

@pdpinch let me know about your thoughts on the updated approach.

pdpinch commented:
> However, we'd still have to modify templates in open edX to add sys admin menu on LMS dashboard.

Can we be specific about what templates would need to be modified? 

We are already overriding templates like [navbar-authenticated.html in the mitx-theme](https://github.com/mitodl/mitx-theme/blob/master/lms/templates/header/navbar-authenticated.html)

Would it be sufficient to override the template?

ziafazal commented:
> Can we be specific about what templates would need to be modified?

Templates where we using `show_sysadmin_dashboard` to render sys admin navigation.

ziafazal commented:
> Would it be sufficient to override the template?

Yes overriding the templates will do. In that case if anyone wants to use sys admin and want sys admin navigation in either learner dashboard or instructor dashboard they'll have to override relevant templates in their custom theme and if they don't have any custom theme then they have setup one or access sysadmin by putting url directly in browser.

HamzaIbnFarooq commented:
According to the above discussion, I have created a PR to move all the sysadmin tab related template code to mitx-theme and methods to sysadmin plugin:
* https://github.com/mitodl/mitx-theme/pull/132
* https://github.com/mitodl/edx-sysadmin/pull/17

pdpinch commented:
Thank you Zia and Hamza. While I see the challenges of the theme-based approach, I think it’s our best option for Lilac which will be cut next Friday. 

I would like to start a community discussion about a proper API for plugin applications to access the header and/or the instructor dashboard UI.

HamzaIbnFarooq commented:
@pdpinch I have created this upstream PR for sysadmin dashboard code removal: https://github.com/edx/edx-platform/pull/27213

Commit:
feat!: remove sysadmin dashboard feature w.r.t 0002-deprecate-sysadmin-dashboard-adr.rst

The sysadmin dashboard feature is converted into a plugable app named edx-sysadmin (https://github.com/mitodl/edx-sysadmin) according to the decisison made at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/dashboard/decisions/0002-deprecate-sysadmin-dashboard-adr.rst. Instances using sysadmin dashboard should use the new plugin from now onwards.

BREAKING CHANGE: sysadmin dashboard is removed

HamzaIbnFarooq commented:
Now, as the upstream PR has been merged (https://github.com/edx/edx-platform/pull/27213), closing this in-house PR.
